tika-parser-modules/tika-parser-pkg-module/src/main/java/org/apache/tika/parser/pkg/PackageParser.java

```diff
@@ -287,6 +284,35 @@ public void parse(
         xhtml.endDocument();
     }
 
+    private ZipArchiveInputStream updateZipArchiveInputStream(ArchiveInputStream ais, Metadata metadata, InputStream stream, String entryEncoding) throws IOException {
+        boolean allowStoredEntriesWithDataDescriptor = false;
+        ZipArchiveInputStream zais;
+        try {
+            readEntries(metadata, ais, null, null, false);
+        } catch (UnsupportedZipFeatureException zfe) {
+            if (zfe.getFeature() == Feature.DATA_DESCRIPTOR) {
+                allowStoredEntriesWithDataDescriptor = true;
+            }
+        } catch (Exception ignored) {}
+        finally {
+            ais.close();
+            stream.reset();
```

Comment:
A `mark(int limit)` should always be called before a `reset()` is called - cause we need to mark the position we want to reset to.

This is not throwing an exception because a `mark(int limit)` was called in `DefaultZipContainerDetector.detect` before this. Besides this I think there're some problems in `TikaInputStream.reset` and some other places where `mark` is not called before `reset`.

I have already finished a fix and would push a PR separatly soon.

Reply:
That makes sense.